cli/src/commands/extensions/mod.rs

```diff
@@ -44,16 +40,67 @@ pub fn command<'a>() -> Command<'a> {
         .subcommand(
             Command::new("new").about("Create a new extension").arg(arg!([PATH]).required(true)),
         )
-        .subcommand(
+        .subcommand(permission_args(
```

Comment:
I don't love the semantics of passing a `Command` to a function that returns `Command` -- it kind of breaks the flow of the builder pattern. 

I suggest we could create an internal trait `CommandExt` with a `with_permission_args()` method that is implemented `for Command`. It is a bit more convoluted but probably also more idiomatic, and other modules requiring permission arguments could just import the trait.

(I won't block the PR on this, as this is just a minor style thing, the current solution is acceptable)

Reply:
I don't see how that would be more idiomatic? And under the hood it would be exactly the same? I really don't see what you'd gain here, it doesn't make any sense to me.

Reply:
Clarity -- I don't understand what `permission_args(command)` means unless I spend some time reading the code.

Reply:
Perhaps a middle ground here would be to use an [`ArgGroup`](https://docs.rs/clap/3.2.15/clap/struct.ArgGroup.html) and add these arguments to each command with `.group(permissions_args())`?

Reply:
I was not aware of `ArgGroup` -- it looks like it's perfectly fit for this use case, I like it.

Reply:
I misunderstood `ArgGroup`. It works by referencing arguments that you have already added to the command. So it doesn't work for this use case.

However, the behavior I was hoping for can be accomplished with `.args(permission_args())` where `permission_args()` returns `Vec<clap::Arg>`